CRM/Core/Resources.php

```diff
@@ -881,4 +882,27 @@ private function resolveFileName(&$fileName, $extName) {
     }
   }
 
+  /**
+   * @param string $url
+   * @return string
+   */
+  public function addCacheCode($url) {
+    parse_str(parse_url($url, PHP_URL_QUERY), $queryParts);
```

Comment:
There was previous discussion on the best approach here from @totten which you can see [here](https://github.com/civicrm/civicrm-core/pull/9977#pullrequestreview-26722269). It might look suspicious but I'm not trying to hide his comments ! I just messed up by opening the PR to the wrong branch.